### PR TITLE
Write an FAQ on how and when to delete assets

### PR DESCRIPTION
**Pull request by natebosch**

### CODE REVIEW

File: docs/builder_author_faq.md [hunk 1]
```diff
@@ -16,3 +16,56 @@ Use `build_to: cache` when:
...
+
+Due to build restrictions - namely that a builder can't read the outputs
+produced by other builders in the same phase - it's sometimes necessary to write
+information to a temporary file in one phase, then read that in a subsequent
+phase, but the intermediate result is not a useful output outside of the build.
+
+Use a `PostProcessBuilder` to "delete" files so they are not included in the
+merged output directory or available through the development server. Note that
+files are never deleted from disk, instead a "delete" by a `PostProcessBuilder`
+acts a filter on what assets can be see in the result of the build. This works
```
Comment (jakemac53) [inline docs/builder_author_faq.md]: s/see/seen

File: docs/builder_author_faq.md [hunk 2]
```diff
@@ -16,3 +16,56 @@ Use `build_to: cache` when:
...
+best if temporary assets have a unique extension. The `FileDeletingBuilder` from
```
Comment (jakemac53) [inline docs/builder_author_faq.md]: nit: consider a new paragraph here

File: docs/builder_author_faq.md [hunk 3]
```diff
@@ -16,3 +16,56 @@ Use `build_to: cache` when:
...
+```
+
+```yaml
+builders:
+  my_builder:
+    import: "package:my_package/builders.dart"
+    builder_factories:
+       - writesTemporary
+       - readsTemporaryWritesPermanent
+    build_extensions: {".dart": [".used_during_build", ".output_for_real"]}
```
Comment (jakemac53) [inline docs/builder_author_faq.md]: nit: remove one of these (keep the one that omits `{` and `[` imo)

File: docs/builder_author_faq.md (hunk 1 above)
Comment (natebosch) [reply docs/builder_author_faq.md]: Done

File: docs/builder_author_faq.md (hunk 2 above)
Comment (natebosch) [reply docs/builder_author_faq.md]: Good call. Done

File: docs/builder_author_faq.md (hunk 3 above)
Comment (natebosch) [reply docs/builder_author_faq.md]: Ha, I though you'd like the second version better which is why I added it, then I forgot to go back and delete the `{}` 😆

**Pull request merged**